hosts.go

```diff
@@ -4,39 +4,49 @@ import (
 	"encoding/json"
 	"net/http"
 
+	"github.com/NebulousLabs/Sia/api"
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
 )
 
-// Make new struct with more fields than what will be returned
-type hostList struct {
-	Hosts  []modules.HostSettings
-	Length int
+// hostDisplayInfo contains the data about each host that will be displayed on
+// the hosts page.
+type hostDisplayInfo struct {
+	IPAddress    modules.NetAddress
+	TotalStorage int64
+	Price        types.Currency
+}
```

Comment:
This is the type that gets serialized and published to users when they query `/hosts.json`. I think there's value in limiting what we output in the spirit of ["Be conservative in what you send, be liberal in what you accept"](https://en.wikipedia.org/wiki/Robustness_principle).